src/Compilers/CSharp/Portable/Emitter/EditAndContinue/CSharpSymbolMatcher.cs

```diff
@@ -490,6 +490,16 @@ public override Symbol VisitNamedType(NamedTypeSymbol sourceType)
                     var typeMap = new TypeMap(otherTypeParameters, otherTypeArguments, allowAlpha: true);
                     return typeMap.SubstituteNamedType(otherDef);
                 }
+                else if (sourceType.IsTupleType)
+                {
+                    var otherDef = (NamedTypeSymbol)this.Visit(originalDef.TupleUnderlyingType);
+                    if ((object)otherDef == null)
+                    {
+                        return null;
+                    }
+
+                    return TupleTypeSymbol.Create(otherDef, sourceType.TupleElementNames);
```

Comment:
>return TupleTypeSymbol.Create(otherDef, sourceType.TupleElementNames); [](start = 20, length = 70)

Is it safe to assume that we always get tuple compatible type symbol back? Could we get in a situation similar to retargeting?